lib/dgram.js

```diff
@@ -344,6 +346,13 @@ Socket.prototype.send = function(buffer,
   if (typeof callback !== 'function')
     callback = undefined;
 
+  if (typeof address === 'function') {
+    callback = address;
+    address = undefined;
+  } else if (typeof address !== 'undefined' && typeof address !== 'string') {
```

Comment:
To be more clear, shouldn't it be `if(address != undefined && typeof address != 'string')` --- that is, any of the 3 values `undefined`, `null`, or String would be acceptable.